draft-kazuho-httpbis-priority.md

```diff
@@ -137,94 +137,36 @@ multiple experiments from independent research have shown that simpler schemes
-avoid changing the agreed upon prioritization scheme.
-
-If there is a prioritization scheme supported by both the client and server,
-then the server's preference order prevails and both peers SHOULD
-only use the agreed upon priority scheme for the remainder of the session.
-The server chooses because it is in the best position to know what
-information from the client is of the most value.
-
-Once the negotiation is complete, endpoints MAY stop sending hop-by-hop
-prioritization signals that were not negotiated in order to conserve bandwidth.
-However, endpoints SHOULD continue sending end-to-end signals (e.g., the
-Priority header field), as that might have meaningful effect to other nodes that
-handle the HTTP message.
-
-## Defined Prioritization Scheme Values
-
-This document defines two prioritization scheme values for use with the
-SETTINGS_PRIORITIES setting.
-
-### H2_TREE {#settings-h2-scheme}
-
-This document defines the priority scheme identifier H2_TREE (8-bit value of 1)
-that indicates support for HTTP/2-style priorities ({{!RFC7540}}, Section 5.3).
-
-The H2_TREE priority scheme identifier MUST NOT be be sent in an HTTP/3 settings
-because there is no defined mapping of this scheme. Endpoints MUST treat receipt
-of H2_TREE as a connection error of type H3_SETTINGS_ERROR.
-
-### URGENCY {#settings-this-scheme}
-
-This document defines the priority scheme identifier URGENCY (8-bit value of 2)
-that indicates support for the extensible priority scheme defined in the present
-document.
-
-An intermediary connecting to a backend server SHOULD declare support for the
-extensible priority scheme when and only when all the requests that are to be
-sent on that backend connection originates from one client-side connection that
-has negotiated the use of the extensible priority scheme (see {{fairness}}).
+## Disabling HTTP/2 Priorities
+
+The problems and insights set out above are motivation for allowing endpoints to
+opt out of using the HTTP/2 priority scheme, in favor of using an alternative
+such as the scheme defined in this specification. Endpoints benefit from
+understanding their peer's intention, so the new
+SETTINGS_DEPRECATE_HTTP2_PRIORITIES is defined. The value of the parameter MUST
+be 0 or 1. Any value other than 0 or 1 MUST be treated as a connection error
+(see {{!RFC7540}}; Section 5.4.1) of type PROTOCOL_ERROR.
+
+Endpoints MUST send this SETTINGS parameter as part of the first SETTINGS frame.
+When the peer receives the first SETTINGS frame, it learns the sender has
+deprecated the HTTP/2 priority scheme if it receives the
+SETTINGS_DEPRECATE_HTTP2_PRIORITIES parameter with the value of 1.
+
+A sender MUST NOT change the SETTINGS_DEPRECATE_HTTP2_PRIORITIES parameter value
+after the first SETTINGS frame. Detection of a change by a receiver MUST be
+treated as a connection error of type PROTOCOL_ERROR.
+
+Until the client receives the SETTINGS frame from the server, the client SHOULD
```

Comment:
"as it sees fit" sounds like this should be a MAY

Reply:
The parenthetical was removed in #116 but it's still a SHOULD, FWIW.

Reply:
I think the current text (SHOULD without "as it sees fit") is fine.

We could change the text to something like: _"SHOULD ..., unless the client has the prior knowledge that the server lacks support for the HTTP/2 prioritization scheme, or that a different prioritization scheme will be used."_ But I am uncertain if we need to be that verbose.